osu.Game/Screens/Play/SkipOverlay.cs

```diff
@@ -143,38 +143,49 @@ protected override void LoadComplete()
         /// </summary>
         public void SkipWhenReady()
         {
-            if (skipQueued) return;
+            if (SkipQueued) return;
 
-            skipQueued = true;
+            SkipQueued = true;
             attemptNextSkip();
 
             void attemptNextSkip() => Scheduler.AddDelayed(() =>
             {
                 if (!button.Enabled.Value)
                 {
-                    skipQueued = false;
+                    attemptNextSkip();
                     return;
                 }
 
                 button.TriggerClick();
-                attemptNextSkip();
+
+                if (progress > 0)
+                    attemptNextSkip();
+                else
+                    SkipQueued = false;
             }, 200);
         }
 
+        private double progress => Math.Max(0, 1 - (gameplayClock.CurrentTime - displayTime) / (fadeOutBeginTime - displayTime));
+
         protected override void Update()
         {
             base.Update();
 
             // This case causes an immediate expire in `LoadComplete`, but `Update` may run once after that.
             // Avoid div-by-zero below.
-            if (fadeOutBeginTime <= displayTime)
-                return;
+            if (fadeOutBeginTime <= displayTime || !gameplayClock.IsRunning)
+            {
+                isClickable = false;
+            }
+            else
+            {
```

Comment:
`isClickable` will now be set to `false` instead of staying `true` when either `progress` would result in division by zero or clock is not running. So button's enable state will be properly set at all times.